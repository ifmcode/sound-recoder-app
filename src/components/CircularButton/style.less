@import (reference) "../../styles/colors.less";

.circular-button {
  position: relative;
  border-radius: 50%;
  background-color: @darkgrey;
  border: 0;
  cursor: pointer;

  .circle-1 {
    background-color: @black;
  }

  .circle-2 {
    background-color: @darkgrey;
  }

  .circle-1, .circle-2 {
    border-radius: 50%;
  }

  .circle-1, .circle-2, .icon, svg {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  //states
  &.is-recording {
    .circle-1 {
      animation: pulse 2s infinite;
    }
  }

  // sizes
  &.small-button {
    width: 28px;
    height: 28px;
    .circle-1 {
      width: 24px;
      height: 24px;
      .circle-2 {
        width: 22px;
        height: 22px;
        .icon {
          width: 20px;
          height: 20px;
          svg {
            width: 14px;
            height: 14px;
          }
        }
      }
    }
  }

  &.medium-button {
    width: 48px;
    height: 48px;
    .circle-1 {
      width: 38px;
      height: 38px;
      .circle-2 {
        width: 34px;
        height: 34px;
        .icon {
          width: 34px;
          height: 34px;
          svg {
            width: 24px;
            height: 24px;
          }
        }
      }
    }
  }

  &.big-button {
    width: 144px;
    height: 144px;
    .circle-1 {
      width: 120px;
      height: 120px;
      .circle-2 {
        width: 106px;
        height: 106px;
        .icon {
          width: 55px;
          height: 55px;
        }
      }
    }
  }
}

@keyframes pulse {
	0% {
		background-color: rgba(32, 200, 192, 1);
	}

	15% {
		background-color: rgba(32, 200, 192, 0.6);
	}

	30% {
		background-color: rgba(32, 200, 192, 0.3);
	}

	50% {
		background-color: rgba(32, 200, 192, 0.1);
	}

	70% {
		background-color: rgba(32, 200, 192, 0.3);
	}

	85% {
		background-color: rgba(32, 200, 192, 0.6);
	}

	100% {
		background-color: rgba(32, 200, 192, 1);
	}
}