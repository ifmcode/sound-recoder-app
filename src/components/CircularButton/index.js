import "./style.less";
import React from 'react';
import Icon from '../Icon';

/**
 * Component of a button that shows a text and it has a custom style
 * @param {String} iconName // possible values are 'settings', 'indicator', 'plane', 'trash' and 'recording'
 * @param {String} iconSize // possible values are 'small", 'medium' and 'big'
 * @param {Boolean} isRecording // Only its used to create pulse animation on big button
 * @param {Function} onClick // Funtion that it call when button is clicked
 * @returns 
 */
const CircularButton = ({iconName, iconSize, isRecording, onClick}) => {
  const buttonClasses = `circular-button ${iconName}-button ${iconSize}-button ${isRecording ? "is-recording" : ""}`;
  return  (
    <button className={buttonClasses} onClick={onClick}>
      <div className="circle-1">
        <div className="circle-2">
          <Icon iconName={iconName}/>
        </div>
      </div>
    </button>
  );
};

export default CircularButton;
