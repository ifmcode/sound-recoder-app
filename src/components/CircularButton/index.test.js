import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react'
import CircularButton from '.';

describe('Testing small CircularButton component', () => {

  let wrapper;
  const onClickFuntion = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <CircularButton
        iconName= {"settings"}
        iconSize= {"small"}
      />
    );
  });
  
  test("should render the CircularButton component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render with correct class', () => {
    const button = screen.getByRole('button');
    expect(button).toHaveClass("small-button");
    expect(button).toHaveClass("settings-button");
  });
})

describe('Testing medium CircularButton component', () => {

  let wrapper;
  const onClickFuntion = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <CircularButton
        iconName= {"trash"}
        iconSize= {"medium"}
        onClick= {onClickFuntion}
      />
    );
  });
  
  test("should render the CircularButton component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render with correct class', () => {
    const button = screen.getByRole('button');
    expect(button).toHaveClass("medium-button");
    expect(button).toHaveClass("trash-button");
  })

  test("should call the onClick function when click in the button", () => {
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(onClickFuntion).toHaveBeenCalledTimes(1);
  });
})

describe('Testing big CircularButton component', () => {

  let wrapper;
  const onClickFuntion = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <CircularButton
        iconName= {"recording"}
        iconSize= {"big"}
        isRecording= {true}
        onClick= {onClickFuntion}
      />
    );
  });
  
  test("should render the CircularButton component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('render with correct class', () => {
    const button = screen.getByRole('button');
    expect(button).toHaveClass("big-button");
    expect(button).toHaveClass("recording-button");
  })

  test("should call the onClick function when click in the button", () => {
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(onClickFuntion).toHaveBeenCalledTimes(1);
  });
})