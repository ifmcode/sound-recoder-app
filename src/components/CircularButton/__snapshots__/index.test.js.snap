// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing big CircularButton component should render the CircularButton component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="circular-button recording-button big-button is-recording"
      >
        <div
          class="circle-1"
        >
          <div
            class="circle-2"
          >
            <div
              class="icon recording-icon"
            >
              <svg>
                recording.svg
              </svg>
            </div>
          </div>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="circular-button recording-button big-button is-recording"
    >
      <div
        class="circle-1"
      >
        <div
          class="circle-2"
        >
          <div
            class="icon recording-icon"
          >
            <svg>
              recording.svg
            </svg>
          </div>
        </div>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing medium CircularButton component should render the CircularButton component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="circular-button trash-button medium-button "
      >
        <div
          class="circle-1"
        >
          <div
            class="circle-2"
          >
            <div
              class="icon trash-icon"
            >
              <svg>
                trash.svg
              </svg>
            </div>
          </div>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="circular-button trash-button medium-button "
    >
      <div
        class="circle-1"
      >
        <div
          class="circle-2"
        >
          <div
            class="icon trash-icon"
          >
            <svg>
              trash.svg
            </svg>
          </div>
        </div>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing small CircularButton component should render the CircularButton component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="circular-button settings-button small-button "
      >
        <div
          class="circle-1"
        >
          <div
            class="circle-2"
          >
            <div
              class="icon settings-icon"
            >
              <svg>
                settings.svg
              </svg>
            </div>
          </div>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="circular-button settings-button small-button "
    >
      <div
        class="circle-1"
      >
        <div
          class="circle-2"
        >
          <div
            class="icon settings-icon"
          >
            <svg>
              settings.svg
            </svg>
          </div>
        </div>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
